src/Crocodealer/Command/Sync.hs

```diff
@@ -16,48 +17,42 @@ data FileStatus
 sync :: UserName -> RepoName -> FileName -> [RepoName] -> IO ()
 sync userName templateRepoName fileName repos = do
     templateFile <- fetchGitHubFile userName templateRepoName fileName
-    putTextLn $ unFileName fileName
-    forM_ repos $ \repo -> do
-        repoFile <- fetchGitHubFile userName repo fileName
-        let fileDiff = calculateDiff templateFile repoFile
-        printFileDiff userName repo fileDiff
+    case templateFile of
+      Nothing -> printFileDiff userName templateRepoName FileDoesNotExist
+      Just f -> do
+        putTextLn $ unFileName fileName
+        forM_ repos $ \repo -> do
+            repoFile <- fetchGitHubFile userName repo fileName
+            case repoFile of
+              Nothing -> printFileDiff userName repo FileDoesNotExist
+              Just r -> do
+                let fileDiff = calculateDiff f r
+                printFileDiff userName repo fileDiff
```

Comment:
This got a little long, but I'm not familiar with a shorter notation for this. :/

Reply:
This is a refactoring I tried and it seems to work, but i'm interested in your opinion:

```haskell
sync :: UserName -> RepoName -> FileName -> [RepoName] -> IO ()
sync userName templateRepoName fileName repos =
    (fetchGitHubFile userName templateRepoName >=> maybe (printErrorMessage templateRepoName)
    (\templateFile -> do
        putTextLn $ unFileName fileName
        forM_ repos $ \repo ->
            (fetchGitHubFile userName repo >=> maybe (printErrorMessage repo)
            (\repoFile -> do
                let fileDiff = calculateDiff templateFile repoFile
                printFileDiff userName repo fileDiff)) fileName)) fileName
  where
    printErrorMessage :: RepoName -> IO()
    printErrorMessage repoName = printFileDiff userName repoName FileDoesNotExist
```

Reply:
I think the current code is fine. I like explicit pattern-matching on the result. And it's actually not that long. One thing you can do to make it shorter is to use `LambdaCase` extension:

```haskell
sync userName templateRepoName fileName repos =
    fetchGitHubFile userName templateRepoName fileName >>= \case
        Nothing -> ...
        Just templateFile -> ...
```

This way the code will become shorter and you'll introduce fewer names to the scope.

Also, let's align this block with 4 spaces, since in our style guide we use 4 spaces 🙂